src/deephaven/plugin/matplotlib/figure_type.py

```diff
@@ -1,9 +1,110 @@
 from io import BytesIO
+from weakref import WeakKeyDictionary, WeakSet
 from matplotlib.figure import Figure
+import matplotlib.pyplot as plt
 from deephaven.plugin.object import Exporter, ObjectType
+from threading import Timer
 
+# Name of the matplotlib figure object that was export
 NAME = "matplotlib.figure.Figure"
 
+# DPI for the figure
+DPI = 144
+
+# Dictionary to store the input tables created for each figure
+_figure_tables = WeakKeyDictionary()
+
+# Track the currently drawing figures, otherwise the stale_callback gets called when we call `savefig`
+_exporting_figures = WeakSet()
+
+def debounce(wait):
+    """Postpone a functions execution until after some time has elapsed
+
+    :type wait: int
+    :param wait: The amount of Seconds to wait before the next call can execute.
+    """
+    def decorator(fun):
+        def debounced(*args, **kwargs):
+            def call_it():
+                fun(*args, **kwargs)
+
+            try:
+                debounced.t.cancel()
+            except AttributeError:
+                pass
+
+            debounced.t = Timer(wait, call_it)
+            debounced.t.start()
+
+        return debounced
+
+    return decorator
+
+# Creates an input table that will update a figures size when the input is set
+# Has three different key value pairs:
+# revision: Increases whenever the figure 'ticks'
+# width: The width of panel displaying the figure
+# height: The height of the panel displaying the figure
+def _make_input_table(figure):
+    from deephaven import new_table
+    from deephaven.column import string_col, int_col
+    import jpy
+    from deephaven.table_listener import _do_locked
+
+    # We need to get the liveness scope so we can run table operations
+    LivenessScope = jpy.get_type('io.deephaven.engine.liveness.LivenessScope')
+    LivenessScopeStack = jpy.get_type('io.deephaven.engine.liveness.LivenessScopeStack')
+    liveness_scope = LivenessScope(True)
+    LivenessScopeStack.push(liveness_scope)
```

Comment:
I believe there are proper wrapers for this now?

Reply:
liveness has been wrapped.

Reply:
Where do I import liveness_scope from? What objects do I need to keep around? What do I do when I'm done with it? What exactly is a liveness scope?

I'm a little confused by it's usage/purpose. Any docs or diagram would be very helpful.

Reply:
Hi, Mike, the liveness scope wrapping was merged early last week. I don't think DevRel has gotten around to document the usage yet. So all we have now is the generated Python doc.
       https://deephaven.io/core/pydoc/code/deephaven.liveness_scope.html.